From 8f10ea7adfee9e87050613cd2199ed32c0412888 Mon Sep 17 00:00:00 2001
From: Christian Hemp <c.hemp@phytec.de>
Date: Wed, 16 Oct 2013 13:34:04 +0200
Subject: [PATCH 28/55] imx6: pfla02: fix nand boot workaround

Add dual lite / Solo muxing for GPIO2_11. We need this for the NAND boot
workaround.

Signed-off-by: Christian Hemp <c.hemp@phytec.de>
---
 arch/arm/boards/phytec-pfla02/board.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/arch/arm/boards/phytec-pfla02/board.c b/arch/arm/boards/phytec-pfla02/board.c
index ed65836..e24b667 100644
--- a/arch/arm/boards/phytec-pfla02/board.c
+++ b/arch/arm/boards/phytec-pfla02/board.c
@@ -41,6 +41,9 @@
 				MX6_PAD_CTL_SPEED_MED | MX6_PAD_CTL_DSE_40ohm | MX6_PAD_CTL_HYS
 #define MX6Q_PAD_SD4_DAT3__GPIO_2_11_PD (_MX6Q_PAD_SD4_DAT3__GPIO_2_11 | MUX_PAD_CTRL(GPIO_2_11_PD_CTL))
 
+#define MX6DL_PAD_SD4_DAT3__GPIO_2_11                                          \
+			IOMUX_PAD(0x0734, 0x034C, 5, 0x0000, 0, GPIO_2_11_PD_CTL)
+
 #define MX6_PHYFLEX_ERR006282	IMX_GPIO_NR(2, 11)
 static void phyflex_err006282_workaround(void)
 {
@@ -49,7 +52,11 @@ static void phyflex_err006282_workaround(void)
 
 	mdelay(20);
 	gpio_set_value(MX6_PHYFLEX_ERR006282, 0);
-	mxc_iomux_v3_setup_pad(MX6Q_PAD_SD4_DAT3__GPIO_2_11_PD);	
+	if (cpu_is_mx6q())
+		mxc_iomux_v3_setup_pad(MX6Q_PAD_SD4_DAT3__GPIO_2_11_PD);
+	else if (cpu_is_mx6dl())
+		mxc_iomux_v3_setup_pad(MX6DL_PAD_SD4_DAT3__GPIO_2_11);
+
 	gpio_direction_input(MX6_PHYFLEX_ERR006282);
 }
 
-- 
1.7.0.4

